File: src/main/java/com/github/nelson54/lostcities/service/CommandService.java
package com.github.nelson54.lostcities.service;

import com.github.nelson54.lostcities.domain.game.Card;
import com.github.nelson54.lostcities.domain.game.Command;
import com.github.nelson54.lostcities.domain.game.Game;
import com.github.nelson54.lostcities.domain.game.Player;
import org.springframework.stereotype.Service;

@Service
public class CommandService {

    public void execute(Game game, Command command) {
        Player player = command.getPlayer();
        Card drew;

        if (command.getPlay() != null) {
            player.play(command.getPlay());
        } else if (command.getDiscard() != null) {
            player.discard(command.getDiscard());

        }

        // Draw from discard
        if (command.getDrawColor() != null) {
            player.draw(command.getDrawColor());
        //Draw from deck
        } else {
            player.draw();
        }

        game.incrementTurn();
    }
}

File: src/main/webapp/game/services/command-service.js
const game = require('../game');

module.exports = class CommandService {

    execute(command) {
        console.log(command);
        axios.put(`/api/game/${game.id}/play`, command)
            .then(function (response) {
                console.log(response);
            })
            .catch(function (error) {
                console.log(error);
            });
    }
}


